django_mongodb/compiler.py

```diff
@@ -98,6 +98,8 @@ def check_query(self):
         if self.query.is_empty():
             raise EmptyResultSet()
         if self.query.distinct:
+            if "datetimefield" in self.query.annotations:
```

Comment:
Do we need to add `.lower()` to `self.query.annotations`?

Reply:
No, this matches the name used by Django: https://github.com/django/django/blob/b79ac89c5799993b459b2248faf0bdd66a9df008/django/db/models/query.py#L1439. I'll add a comment about that.

This is a heuristic since technically a user could use an annotation of the same name. It seems unlikely, and it's unclear to what extent we'll support annotations anyway.